arch/Mips/MipsInstPrinter.c

```diff
@@ -1,261 +1,280 @@
-		case Mips_FCOND_T:   return "f";
-		case Mips_FCOND_UN:
-		case Mips_FCOND_OR:  return "un";
-		case Mips_FCOND_OEQ:
-		case Mips_FCOND_UNE: return "eq";
-		case Mips_FCOND_UEQ:
-		case Mips_FCOND_ONE: return "ueq";
-		case Mips_FCOND_OLT:
-		case Mips_FCOND_UGE: return "olt";
-		case Mips_FCOND_ULT:
-		case Mips_FCOND_OGE: return "ult";
-		case Mips_FCOND_OLE:
-		case Mips_FCOND_UGT: return "ole";
-		case Mips_FCOND_ULE:
-		case Mips_FCOND_OGT: return "ule";
-		case Mips_FCOND_SF:
-		case Mips_FCOND_ST:  return "sf";
-		case Mips_FCOND_NGLE:
-		case Mips_FCOND_GLE: return "ngle";
-		case Mips_FCOND_SEQ:
-		case Mips_FCOND_SNE: return "seq";
-		case Mips_FCOND_NGL:
-		case Mips_FCOND_GL:  return "ngl";
-		case Mips_FCOND_LT:
-		case Mips_FCOND_NLT: return "lt";
-		case Mips_FCOND_NGE:
-		case Mips_FCOND_GE:  return "nge";
-		case Mips_FCOND_LE:
-		case Mips_FCOND_NLE: return "le";
-		case Mips_FCOND_NGT:
-		case Mips_FCOND_GT:  return "ngt";
+	case Mips_FCOND_F:
+	case Mips_FCOND_T:
+		return "f";
+	case Mips_FCOND_UN:
+	case Mips_FCOND_OR:
+		return "un";
+	case Mips_FCOND_OEQ:
+	case Mips_FCOND_UNE:
+		return "eq";
+	case Mips_FCOND_UEQ:
+	case Mips_FCOND_ONE:
+		return "ueq";
+	case Mips_FCOND_OLT:
+	case Mips_FCOND_UGE:
+		return "olt";
+	case Mips_FCOND_ULT:
+	case Mips_FCOND_OGE:
+		return "ult";
+	case Mips_FCOND_OLE:
+	case Mips_FCOND_UGT:
+		return "ole";
+	case Mips_FCOND_ULE:
+	case Mips_FCOND_OGT:
+		return "ule";
+	case Mips_FCOND_SF:
+	case Mips_FCOND_ST:
+		return "sf";
+	case Mips_FCOND_NGLE:
+	case Mips_FCOND_GLE:
+		return "ngle";
+	case Mips_FCOND_SEQ:
+	case Mips_FCOND_SNE:
+		return "seq";
+	case Mips_FCOND_NGL:
+	case Mips_FCOND_GL:
+		return "ngl";
+	case Mips_FCOND_LT:
+	case Mips_FCOND_NLT:
+		return "lt";
+	case Mips_FCOND_NGE:
+	case Mips_FCOND_GE:
+		return "nge";
+	case Mips_FCOND_LE:
+	case Mips_FCOND_NLE:
+		return "le";
+	case Mips_FCOND_NGT:
+	case Mips_FCOND_GT:
+		return "ngt";
 	}
+	assert(0 && "Impossible condition code!");
+	return "";
 }
 
-static void printRegName(SStream *OS, unsigned RegNo)
-{
-	SStream_concat(OS, "$%s", getRegisterName(RegNo));
-}
+const char *Mips_LLVM_getRegisterName(unsigned RegNo, bool noRegName);
 
-void Mips_printInst(MCInst *MI, SStream *O, void *info)
+static void printRegName(MCInst *MI, SStream *OS, MCRegister Reg)
 {
-	char *mnem;
+	int syntax_opt = MI->csh->syntax;
+	if (!(syntax_opt & CS_OPT_SYNTAX_NO_DOLLAR)) {
+		SStream_concat1(OS, '$');
+	}
+	SStream_concat0(OS, Mips_LLVM_getRegisterName(Reg, syntax_opt & CS_OPT_SYNTAX_NOREGNAME));
+}
 
-	switch (MCInst_getOpcode(MI)) {
-		default: break;
-		case Mips_Save16:
-		case Mips_SaveX16:
-		case Mips_Restore16:
-		case Mips_RestoreX16:
-			return;
+void Mips_LLVM_printInst(MCInst *MI, uint64_t Address, SStream *O) {
+	bool useAliasDetails = map_use_alias_details(MI);
+	if (!useAliasDetails) {
+		SStream_Close(O);
+		printInstruction(MI, Address, O);
+		SStream_Open(O);
+		map_set_fill_detail_ops(MI, false);
 	}
```

Comment:
Please add at least one test for the "alias instruction with real details". Just to make sure it works as expected.